File: sdk/python/overseer_sdk/styles.py
"""Overseer design system for Python plugins.

Mirrors the lipgloss palette defined in cli/internal/tui/styles.go.
Uses rich for rendering — add `rich` to your plugin's dependencies.

Usage:
    from rich.console import Console
    from overseer_sdk.styles import section_header, warn_line, STYLE_OK

    console = Console()
    console.print(section_header("GitHub", "3 open PRs"))
    console.print(warn_line("auth", "token expired"))
"""

from rich.style import Style
from rich.text import Text

# --- Palette --------------------------------------------------------------- #
# 256-colour codes to match the lipgloss palette in styles.go.

STYLE_HEADER = Style(color="color(99)", bold=True)   # purple — section titles
STYLE_ACCENT = Style(color="color(212)")              # pink   — keys, channels, usernames
STYLE_OK     = Style(color="color(82)",  bold=True)  # green  — success
STYLE_WARN   = Style(color="color(214)", bold=True)  # amber  — warnings
STYLE_ERROR  = Style(color="color(196)", bold=True)  # red    — errors
STYLE_MUTED  = Style(color="color(240)")              # dark   — hints, empty state
STYLE_DIM    = Style(color="color(245)")              # grey   — secondary info
STYLE_NORMAL = Style(color="color(252)")              # light  — body text

# --- Helpers --------------------------------------------------------------- #


def section_header(label: str, badge: str = "") -> Text:
    """Render a styled section header with an optional badge.

    Mirrors tui.SectionHeader — output example:
        ▸ GitHub  ·  3 open
    """
    t = Text("▸ " + label, style=STYLE_HEADER)
    if badge:
        t.append("  ·  " + badge, style=STYLE_MUTED)
    return t


def warn_line(label: str, msg: str) -> Text:
    """Render a warning line.

    Mirrors tui.WarnLine — output example:
        ⚠  auth: token expired
    """
    t = Text("⚠  " + label + ":", style=STYLE_WARN)
    t.append(" " + msg, style=STYLE_MUTED)
    return t

File: sdk/typescript/src/styles.ts
/**
 * Overseer design system for TypeScript plugins.
 *
 * Mirrors the lipgloss palette defined in cli/internal/tui/styles.go.
 * Uses chalk for rendering — add `chalk` to your plugin's dependencies.
 *
 * @example
 *   import { styles, sectionHeader, warnLine } from "@overseer/sdk";
 *
 *   console.log(sectionHeader("GitHub", "3 open PRs"));
 *   console.log(warnLine("auth", "token expired"));
 */

import chalk from "chalk";

// --- Palette --------------------------------------------------------------- //
// 256-colour codes to match the lipgloss palette in styles.go.

export const styles = {
  /** purple — section titles */
  header: (text: string) => chalk.ansi256(99).bold(text),
  /** pink — keys, channels, usernames */
  accent: (text: string) => chalk.ansi256(212)(text),
  /** green — success */
  ok: (text: string) => chalk.ansi256(82).bold(text),
  /** amber — warnings */
  warn: (text: string) => chalk.ansi256(214).bold(text),
  /** red — errors */
  error: (text: string) => chalk.ansi256(196).bold(text),
  /** dark grey — hints, empty state */
  muted: (text: string) => chalk.ansi256(240)(text),
  /** grey — secondary info */
  dim: (text: string) => chalk.ansi256(245)(text),
  /** light — body text */
  normal: (text: string) => chalk.ansi256(252)(text),
};

// --- Helpers --------------------------------------------------------------- //

/**
 * Render a styled section header with an optional badge.
 *
 * Mirrors tui.SectionHeader — output example:
 *   ▸ GitHub  ·  3 open
 */
export function sectionHeader(label: string, badge?: string): string {
  let s = styles.header("▸ " + label);
  if (badge) s += "  " + styles.muted("·  " + badge);
  return s;
}

/**
 * Render a warning line.
 *
 * Mirrors tui.WarnLine — output example:
 *   ⚠  auth: token expired
 */
export function warnLine(label: string, msg: string): string {
  return styles.warn("⚠  " + label + ":") + " " + styles.muted(msg);
}

/**
 * Render a success line.
 *
 * Output example:
 *   ✓  label: message
 */
export function okLine(label: string, msg = ""): string {
  let s = styles.ok("✓  " + label);
  if (msg) s += styles.muted(": " + msg);
  return s;
}

/**
 * Render an error line.
 *
 * Output example:
 *   ✗  label: message
 */
export function errorLine(label: string, msg = ""): string {
  let s = styles.error("✗  " + label);
  if (msg) s += styles.muted(": " + msg);
  return s;
}
